imap_processing/ultra/l1c/ultra_l1c_pset_bins.py

```diff
@@ -174,3 +209,123 @@ def get_pointing_frame_exposure_times(
         exposure = cdf_file.varget(f"dps_grid{sensor}") * n_spins
 
     return exposure
+
+
+@ensure_spice
+@typing.no_type_check
+def get_helio_exposure_times(
+    time: np.ndarray,
+    sc_exposure: np.ndarray,
+) -> np.ndarray:
+    """
+    Compute a 3D array of the exposure in the helio frame.
+
+    Parameters
+    ----------
+    time : np.ndarray
+        Median time of pointing.
+    sc_exposure : np.ndarray
+        Spacecraft exposure.
+
+    Returns
+    -------
+    exposure_3d : np.ndarray
+        A 3D array with dimensions (az, el, energy).
+
+    Notes
+    -----
+    These calculations are performed once per pointing.
+    """
+    # Get bins and midpoints.
+    energy_bin_edges, energy_midpoints = build_energy_bins()
+    az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints = (
+        build_spatial_bins()
+    )
+    # Initialize the exposure grid.
+    exposure_3d = np.zeros(
+        (len(el_bin_midpoints), len(az_bin_midpoints), len(energy_midpoints))
+    )
+
+    # Create a 3D Cartesian grid from spherical coordinates
+    # using azimuth and elevation midpoints.
+    az_grid, el_grid = np.meshgrid(az_bin_midpoints, el_bin_midpoints[::-1])
+
+    # Radial distance.
+    r = np.ones(el_grid.shape)
+    x, y, z = spherical_to_cartesian(r, np.radians(az_grid), np.radians(el_grid))
+
+    # Reshape and combine the Cartesian coordinates into a 2D array.
+    cartesian = np.vstack(
+        [x.flatten(order="F"), y.flatten(order="F"), z.flatten(order="F")]
+    )
+
+    # Spacecraft velocity in the pointing (DPS) frame wrt heliosphere.
+    state, lt = spice.spkezr("IMAP", time, "IMAP_DPS", "NONE", "SUN")
```

Comment:
There is a common function for this: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/spice/geometry.py#L86
```suggestion
    state = imap_state(time, SpiceFrame.IMAP_DPS, SpiceBody.Sun)
```